lib/CLI/Parser.hs

```diff
@@ -38,15 +38,28 @@ import Coffer.Path (Path, mkPath, EntryPath, mkEntryPath)
 
 {-# ANN module ("HLint: ignore Use <$>" :: Text) #-}
 
-parserInfo :: ParserInfo SomeCommand
-parserInfo =
-  info (parser <**> helper) $
+parserInfo :: Text -> ParserInfo Options
+parserInfo defaultBackend =
+  info (parser defaultBackend <**> helper) $
     fullDesc
     <> progDesc "TODO: coffer description goes here"
     <> header "TODO: coffer description goes here"
 
-parser :: Parser SomeCommand
-parser =
+parser :: Text -> Parser Options
+parser defaultBackend = Options <$> optionalBackendParser <*> commandParser
+  where
+    optionalBackendParser :: Parser Text
+    optionalBackendParser = 
+      option str $ mconcat
+        [ long "backend"
+        , short 'b'
+        , metavar "BACKEND"
+        , value defaultBackend
+        , help "Backend with which actions will be performed. If not specified, then main backend would be chosen" 
+        ]
```

Comment:
I thought the consensus was that the backend should be part of the path (and not a separate cli option)?

Reply:
Supported both variants (`[-b|--backend]` for `find` and `rename` and hash-separated for other commands).

Reply:
Why should we support both variants?

Reply:
I think it's weird to use `QualifiedPath` in `rename` command because we can perform 
````
coffer rename first#entry1 second#entry2
````

In `find` command path is optional, so we can clarify the backend we are working with without providing the path.

Reply:
> I think it's weird to use QualifiedPath in rename command because we can perform
> `coffer rename first#entry1 second#entry2`

Why is that weird? It seems ok to me. It's not that different from `coffer copy first#entry1 second#entry2`, which we want to support.

----


> In `find` command path is optional, so we can clarify the backend we are working on without providing the path.

Richard addressed that point here: https://github.com/serokell/coffer/issues/16#issuecomment-1048858199

> Why not `coffer find backend# search_term`?

Reply:
Done